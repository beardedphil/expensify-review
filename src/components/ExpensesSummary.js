import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    return (
        <div>
            <h3>Viewing {props.expenseCount} expense{props.expenseCount === 1 ? '' : 's'} totalling ${props.expensesTotal}</h3>
        </div>
    );
}

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: expenses.length,
        expensesTotal: selectExpensesTotal(expenses) / 100
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
