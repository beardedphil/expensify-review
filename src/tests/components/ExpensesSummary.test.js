import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0} />)
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={23.45} />)
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={94.34} />);
    expect(wrapper).toMatchSnapshot();
});
