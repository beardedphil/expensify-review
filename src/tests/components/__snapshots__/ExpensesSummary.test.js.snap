// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    2
     expense
    s
     totalling $
    94.34
  </h3>
</div>
`;

exports[`should correctly render ExpensesSummary with no expenses 1`] = `
<div>
  <h3>
    Viewing 
    0
     expense
    s
     totalling $
    0
  </h3>
</div>
`;

exports[`should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     expense
     totalling $
    23.45
  </h3>
</div>
`;
